### PR TITLE
Null safety migration

### PR DESCRIPTION
**Pull request by FZambia**
Null safety migration

### CODE REVIEW

File: lib/src/subscription.dart [hunk 1]
```diff
@@ -76,7 +74,7 @@ class SubscriptionImpl implements Subscription {
   @override
   void subscribe() {
     _state = _SubscriptionState.subscribed;
-    if (!_client.connected) {
+    if (!_client.connected!) {
```
Comment (LevinGermann) [inline lib/src/subscription.dart:77]: Since the `connected` property can't be null, '!' is not needed here.

File: lib/src/subscription.dart [hunk 2]
```diff
@@ -96,7 +94,7 @@ class SubscriptionImpl implements Subscription {
     }
     _state = _SubscriptionState.unsubscribed;
 
-    if (!_client.connected) {
+    if (!_client.connected!) {
```
Comment (LevinGermann) [inline lib/src/subscription.dart:97]: Since the `connected` property can't be null, '!' is not needed here.

File: lib/src/subscription.dart (hunk 1 above)
Comment (FZambia) [reply lib/src/subscription.dart]: Fixed

File: lib/src/subscription.dart (hunk 2 above)
Comment (FZambia) [reply lib/src/subscription.dart]: Fixed, thanks!

File: lib/src/client.dart [hunk 3]
```diff
@@ -10,7 +10,7 @@ import 'proto/client.pb.dart';
 import 'subscription.dart';
 import 'transport.dart';
 
-Client createClient(String url, {ClientConfig config}) => ClientImpl(
+Client createClient(String url, {ClientConfig? config}) => ClientImpl(
```
Comment (FZambia) [inline lib/src/client.dart:13]: ClientConfig  should not be nullable I suppose

File: lib/src/error.dart [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class Error {
   final int code;
-  final String message;
+  final String? message;
```
Comment (FZambia) [inline lib/src/error.dart:3]: should not be nullable

File: lib/src/events.dart [hunk 5]
```diff
@@ -5,7 +5,7 @@ import 'proto/client.pb.dart' as proto;
 class PrivateSubEvent {
   PrivateSubEvent(this.clientID, this.channel);
 
-  final String clientID;
+  final String? clientID;
```
Comment (FZambia) [inline lib/src/events.dart:8]: should not be nullable

File: lib/src/events.dart [hunk 6]
```diff
@@ -33,8 +33,8 @@ class ConnectEvent {
 class DisconnectEvent {
   DisconnectEvent(this.reason, this.shouldReconnect);
 
-  final String reason;
-  final bool shouldReconnect;
+  final String? reason;
```
Comment (FZambia) [inline lib/src/events.dart:36]: should not be nullable

File: lib/src/events.dart [hunk 7]
```diff
@@ -33,8 +33,8 @@ class ConnectEvent {
...
+  final bool? shouldReconnect;
```
Comment (FZambia) [inline lib/src/events.dart:37]: should not be nullable

File: lib/src/events.dart [hunk 8]
```diff
@@ -128,7 +128,7 @@ class SubscribeSuccessEvent {
 class SubscribeErrorEvent {
   SubscribeErrorEvent(this.message, this.code);
 
-  final String message;
+  final String? message;
```
Comment (FZambia) [inline lib/src/events.dart:131]: should not be nullable

File: lib/src/transport.dart [hunk 9]
```diff
@@ -103,13 +102,13 @@ class Transport implements GeneratedMessageSender {
...
 
-    final data = _commandEncoder.convert(command);
+    final List<int> data = _commandEncoder.convert(command);
 
     if (_socket == null) {
       throw centrifuge.ClientDisconnectedError;
     }
 
-    _socket.add(data);
+    _socket!.add(data);
```
Comment (FZambia) [inline lib/src/transport.dart:111]: what will happen here is _socket is null?

**Pull request merged**